src/plugins/advanced_settings/public/management_app/advanced_settings.tsx

```diff
@@ -145,9 +143,54 @@ export class AdvancedSettingsComponent extends Component<
   }
 
   componentWillUnmount() {
-    if (this.uiSettingsSubscription) {
-      this.uiSettingsSubscription.unsubscribe();
+    this.uiSettingsSubscription?.unsubscribe?.();
+    this.unregister?.();
+  }
+
+  private getQuery(queryString: string): Query {
+    return Query.parse(queryString ? getAriaName(queryString) : '');
+  }
+
+  private getQueryText(pathname?: string, search?: string): string {
+    const match = pathname
+      ? matchPath<RouteParams>(pathname, {
+          path: this.props.match?.path,
+          exact: this.props.match?.isExact,
+        })
+      : this.props.match;
+
+    const routeQuery = match?.params[QUERY] ?? '';
+
+    if (routeQuery) return routeQuery;
+
+    return new URLSearchParams(search ?? this.props.location.search).get(QUERY) ?? '';
```

Comment:
I recommend you to use **url.encodeUriQuery()** from `import { url } from '../../../kibana_utils/public/';` instead of **URLSearchParams** which we use across the whole kibana. This utility make our URL's a little bit more user friendly.

Please compare: 
URLSearchParams:  `settings?query=category%3A%28discover+or+notifications%29`
url.encodeUriQuery: `settings?query=category:(discover or notifications)`

Reply:
I'm not sure what you mean here. I don't see how `url.encodeUriQuery` can replace `URLSearchParams` to parse and extract a search param. `URLSearchParams` is used in 25 places across kibana.

Reply:
Maybe I highlighted a wrong line but in URL I expect to see `settings?query=category:(discover or notifications)` instead of `settings?query=category%3A%28discover+or+notifications%29`

Reply:
Ok see 3f12a65